### PR TITLE
Support TLS for okhttp benchmark.

### PR DESCRIPTION
**Pull request by madongfly**
And the first run result:
Channels:                       4
Outstanding RPCs per Channel:   10
Server Payload Size:            0
Client Payload Size:            0
50%ile Latency (in micros):     2057
90%ile Latency (in micros):     2821
95%ile Latency (in micros):     3117
99%ile Latency (in micros):     3827
99.9%ile Latency (in micros):   7659
Maximum Latency (in micros):    127487
QPS:                            18344

@ejona86, please take a look, thanks!

### CODE REVIEW

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java [hunk 1]
```diff
@@ -129,10 +131,27 @@ static SimpleRequest newRequest(ClientConfiguration config) {
...
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        } else {
+          factory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+        }
+        builder.sslSocketFactory(factory);
+      }
+
+
```
Comment (ejona86) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java]: Remove some of these empty lines?

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java [hunk 2]
```diff
@@ -129,10 +131,27 @@ static SimpleRequest newRequest(ClientConfiguration config) {
...
-      return OkHttpChannelBuilder
+      OkHttpChannelBuilder builder = OkHttpChannelBuilder
           .forAddress(addr.getHostName(), addr.getPort())
-          .executor(config.directExecutor ? MoreExecutors.newDirectExecutorService() : null)
-          .build();
+          .executor(config.directExecutor ? MoreExecutors.newDirectExecutorService() : null);
+      if (config.tls) {
+        SSLSocketFactory factory;
+        if (config.testca) {
+          builder.overrideHostForAuthority(TestUtils.TEST_SERVER_HOST);
```
Comment (ejona86) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java]: You shouldn't hard-code the override. Instead, pass `addr.getAddress().getHostAddress()` to `forAddress()` and always call `overrideHostForAuthority()` with `addr.getHostString()`. Yes, getHostString() is only in Java 7, but it doesn't seem bad to depend on Java 7 here.

Comment (madongfly) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java]: I believe Zhouhan is working on running this benchmark on Android, and she'll need this class.

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java (hunk 1 above)
Comment (madongfly) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java]: Done.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

File: benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java (hunk 2 above)
Comment (ejona86) [inline benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java]: Hmm... It looks like this is hard-coded for Netty as well. I guess we don't have a flag to override it to something else. Very well.

**Comment by ejona86** [comment]
@madongfly LGTM

**Pull request merged**